tracim/tracim/controllers/page.py

```diff
@@ -0,0 +1,103 @@
+import tg
+from tg import expose
+from tg import tmpl_context
+from preview_generator.manager import PreviewManager
+
+from tracim.config.app_cfg import CFG
+from tracim.controllers import TIMRestController
+from tracim.lib.content import ContentApi
+from tracim.model.data import ContentType
+
+__all__ = ['PagesController']
+
+
+class PagesController(TIMRestController):
+
+    @expose()
+    def _default(self):
+        return '<h2> Error Loading Page</h2>'
+
+    @expose()
+    def get_all(self, *args, **kwargs):
+        file_id = int(tg.request.controller_state.routing_args.get('file_id'))
+        return 'all the pages of document {}'.format(file_id)
+
+    @expose(content_type='image/jpeg')
+    def get_one(self,
+                page_id: str='-1',
+                revision_id: str=None,
+                size: int=300,
+                *args, **kwargs):
+        file_id = int(tg.request.controller_state.routing_args.get('file_id'))
+        page = int(page_id)
+        revision_id = int(revision_id) if revision_id != 'latest' else None
+        cache_path = CFG.get_instance().PREVIEW_CACHE
+        preview_manager = PreviewManager(cache_path, create_folder=True)
+        user = tmpl_context.current_user
+        content_api = ContentApi(user,
+                                 show_archived=True,
+                                 show_deleted=True)
+        if revision_id:
+            file_path = content_api.get_one_revision_filepath(revision_id)
+        else:
+            file = content_api.get_one(file_id, self._item_type)
+            file_path = content_api.get_one_revision_filepath(file.revision_id)
+        path = preview_manager.get_jpeg_preview(file_path=file_path,
+                                                page=page,
+                                                height=size,
+                                                width=size)
```

Comment:
Add a FIXME for moving this "intelligence" into lib/content.py
Add a bug for processing this fixme (add it to the "technical improvments" milestone)